src/store/flow/utils/resourceHelpers.ts

```diff
@@ -165,3 +165,14 @@ export function findBlockRelatedResourcesUuids({block}: {block: IBlock}): IResou
 
   return resources
 }
+
+/**
+ * Compute resource index (cell index) for a table having X languages and Y modes
+ *
+ * @param languageIndex
+ * @param modeIndex
+ * @param supportedModeLength
+ */
+export function computeResourceIndex(languageIndex: number, modeIndex: number, supportedModeLength: number): number {
+  return languageIndex * supportedModeLength + modeIndex
+}
```

Comment:
I'd like to suggest renaming `supportedModeLength` for extra clarity. 
For myself, it sounds like `'USSD'.length === 4` 😃 

```suggestion
export function computeResourceIndex(languageIndex: number, modeIndex: number, suppoertedModesCount: number): number {
  return languageIndex * suppoertedModesCount + modeIndex
}
```

Reply:
Got you, thanks. Renamed it in a different commit as this suggestion has a type ;)